### PR TITLE
[DOCS] Sort option for the grok patterns endpoint

### PR DESCRIPTION
**Pull request by danhermann**
Documents the new sort option for the grok patterns endpoint added in #61671.

### CODE REVIEW

**Comment by danhermann** [comment]
@elasticmachine update branch

File: docs/reference/ingest/processors/grok.asciidoc [hunk 1]
```diff
@@ -305,6 +305,32 @@ The above request will return a response body containing a key-value representat
 --------------------------------------------------
 // NOTCONSOLE
 
+The request has an optional boolean parameter for returning the patterns sorted
+by name. By default, the patterns are returned in the order they are read from
+disk which preserves groupings of related patterns (e.g., all patterns
+related to parsing Linux syslog lines).
```
Comment (jrodewig) [inline docs/reference/ingest/processors/grok.asciidoc:311]:
```suggestion
By default, the API returns patterns in the order they are read from disk. This
sort order preserves groupings of related patterns. For example, all patterns
related to parsing Linux syslog lines stay grouped together.

You can use the optional boolean `s` query parameter to sort returned patterns
by key name instead.
```

File: docs/reference/ingest/processors/grok.asciidoc [hunk 2]
```diff
@@ -305,6 +305,32 @@ The above request will return a response body containing a key-value representat
...
+
+[source,console]
+--------------------------------------------------
+GET _ingest/processor/grok?s
+--------------------------------------------------
+
+The above request will return a response body containing a key-value
+representation of the built-in patterns dictionary sorted by key name.
```
Comment (jrodewig) [inline docs/reference/ingest/processors/grok.asciidoc:319]:
```suggestion
The API returns the following response.
```

**Review by jrodewig — APPROVED** [review]
LGTM. I left some suggestions to reorder the sentences, avoid future tense, etc.
Feel free to disregard if wanted. Thanks!

**Pull request merged**